internal/controllers/machinedeployment/mdutil/util.go

```diff
@@ -188,6 +188,11 @@ func getIntFromAnnotation(ms *clusterv1.MachineSet, annotationKey string, logger
 	return int32(intValue), true
 }
 
+// revisionHistoryAnnotation maintains the history of all old revisions that a machine set has served for a machine deployment.
+//
+// Deprecated: This annotation is deprecated and is going to be removed in the next release.
+const revisionHistoryAnnotation = "machinedeployment.clusters.x-k8s.io/revision-history"
```

Comment:
This annotation will be removed in a follow-up (tracked in https://github.com/kubernetes-sigs/cluster-api/issues/10852)

Main goal of this PR is to drop the user-facing revision history feature